crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -1,4 +1,80 @@
-//! The workflow and queue consumer for sys validation
+//! Holochain workflow to validate all incoming DHT operations with an
+//! app-defined validation function.
+//!
+//! Triggered by system validation, this workflow iterates over a list of
+//! [`DhtOp`]s to be validated, validates each op, updates its validation status
+//! in the database accordingly, and triggers op integration if necessary.
+//!
+//! # Sequential validation
+//!
+//! Ops are validated in sequence based on their op type and the timestamp they
+//! were authored (see [`OpOrder`] and [`OpNumericalOrder`]). Validating one op
+//! after the other with this ordering ensures that ops that depend on earlier
+//! ops will be validated after the earlier ops. An example is an incoming delete
+//! op that depends on a create op. Validated in order of their authoring, the
+//! create op is validated first, followed at some stage by the delete op. If
+//! the validation function references the original action when validating
+//! delete ops, the create op will have been validated and is available in the
+//! database. Otherwise the delete op could not be validated and its dependency,
+//! the create op, would be awaited first.
+//!
+
+// This does not apply right now, because all actions are written to the database
```

Comment:
What specifically -- the preceding paragraph or the following paragraph? (I know this line doesn't make it into the documentation, so it may not matter much, but I'm not quite sure what it's referring to)

Reply:
The preceding paragraph. Updated.